app/services/vine/voucher_validator_service.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module Vine
+  class VoucherValidatorService
+    attr_reader :voucher_code, :errors
+
+    def initialize(voucher_code:, enterprise:)
+      @voucher_code = voucher_code
+      @enterprise = enterprise
+      @errors = {}
+    end
+
+    def validate
+      if vine_settings.nil?
+        errors[:vine_settings] = I18n.t("vine_voucher_validator_service.errors.vine_settings")
+        return nil
+      end
+
+      response = call_vine_api
+
+      if !response.success?
+        handle_errors(response)
+        return nil
+      end
```

Comment:
I reworked the error handling to use `Faraday::Response::RaiseError`.
`VineApiService` logs any Client or Server error and re raise the exception so it can be handle by the calling service. 
Error handling starts here : https://github.com/openfoodfoundation/openfoodnetwork/pull/12949/commits/065a9a183259e936de6bb2e2df00a5ff5dec8304